torch/utils/data/datapipes/utils/common.py

```diff
@@ -16,6 +18,35 @@
 ]
 
 
+def ensure_map_fn_works(fn: Callable, input_col: Optional[Union[int, tuple, list]]):
+    """
+    Checks that function used in a map style datapipe works with the input column
+
+    Args:
+        fn: The function to check.
+        input_col: The input column to check.
+    Returns:
+        None.
+    Raises:
+        TypeError: If the function is not compatible with the input column.
+    """
+    sig = inspect.signature(fn)
+
```

Comment:
First, the logic can be simplified here by getting length of `input_col`.
- 1 for int
- `len(input_col)` for list or tuple

Second, as I said, comparing `len(sig.parameters)` is not enough. There can be default value for each argument. We should validate if `len(sig.parameters)` is larger than the length of `input_col`. The remaining parameters should have default value.

Reply:
I was able to use inspect to pull out the non-default parameters, and I simply validate if the number of non-default parameters is `len(input_col)`. Is that what you have in mind?

Reply:
It still doesn't cover all the cases. Let's say we have a function like `def fn(arg1, arg2=2, arg3=3)`. The number of non-default parameters is 1. But, we allow users to put any number of `input_col` from 1 to 3.

Reply:
Oh, so I should just check if the number of non-default parameters is <= `len(input_col)`?